src/Components/Scorecard.js

```diff
@@ -1,150 +1,114 @@
+  }
+
+  const changeScore = (id, score) => {
+    let player = scorecard.find(player => player.id === id)
+
+    player.score = score
+    const otherPlayers = scorecard.filter(player => player.id !== id)
+    setScorecard([...otherPlayers, player])
+    setCurrentHole(currentHole)
+  }
+
+  const nextHole = () => {
+    const hole = courseInfo.currentCourse.layout.holes.find(hole => hole.hole_number === currentHole.number + 1)
+    if(hole.hole_number === courseInfo.currentCourse.layout.holes.length){
       setInProgress(false)
     }
     setCurrentHole({
       number: hole.hole_number,
-      players: updatePlayerScores(prevHoleScores),
       par: hole.par,
       distance: hole.distance
     })
   }
 
-  const updatePlayerScores = (prevHoleScores) => {
-    const updatedPlayers = {}
-    prevHoleScores.forEach(player => {
-      updatedPlayers[player.player_id] = {
-        name: currentHole.players[player.player_id].name,
-        id: currentHole.players[player.player_id].id,
-        score: 0,
-        totalScore: player.score + currentHole.players[player.player_id].totalScore
-      }
-    })
-    return updatedPlayers
-  }
-
   const postHoleScores = () => {
-    const currentPlayers = Object.keys(currentHole.players)
-    const playerScores = currentPlayers.map(player_id => {
-      return {
-        player_id: currentHole.players[player_id].id,
-        strokes: currentHole.players[player_id].score
-      }
-    })
-
     const body = {
       hole: {
         hole_number: currentHole.number,
-        player_scores: playerScores
+        player_scores: scorecard.map(player => {
+          return {
+            player_id: player.id,
+            strokes: player.score === 0 ? 
+              currentHole.par : 
+              player.score
+          }
+        })
       }
     }
-    scoreAHole(body, scorecard.roundId)
-    .then(data => nextHole(data.scores))
-    .catch(error => setErrorMessage(error))
-  }
-
-  const postFinalScores = () => {
-    const currentPlayers = Object.keys(currentHole.players)
-    const playerScores = currentPlayers.map(player_id => {
-      return {
-        player_id: currentHole.players[player_id].id,
-        strokes: currentHole.players[player_id].score
-      }
+    scoreAHole(body, courseInfo.roundId)
+    .then(data => {
+      setScores(data.scores)
+      inProgress ? nextHole() : endCurrentRound()
     })
-
-    const body = {
-      hole: {
-        hole_number: currentHole.number,
-        player_scores: playerScores
-      }
-    }
-    scoreAHole(body, scorecard.roundId)
-    .then(() =>  endCurrentRound())
     .catch(error => setErrorMessage(error))
   }
 
   const endCurrentRound = () => {
-    endRound(scorecard.roundId)
-    .then(data => {
-      setScorecard({
-        ...scorecard,
-        final: data
-      })
-      navigate('/round-overview')
-    })
+    endRound(courseInfo.roundId)
+    .then(data => navigate('/round-overview', {state: data}))
```

Comment:
The documentation on this is not very helpful. It took some tinkering to figure out that we could pass the data as an argument in navigate so that we could render the data on the next page.